app/lib/features/chat_ng/widgets/events/chat_event.dart

```diff
@@ -0,0 +1,199 @@
+import 'package:acter_avatar/acter_avatar.dart';
+import 'package:flutter/material.dart';
+import 'package:acter_flutter_sdk/acter_flutter_sdk_ffi.dart'
+    show RoomMessage, RoomVirtualItem, RoomEventItem;
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:logging/logging.dart';
+
+final _log = Logger('a3::chat::widgets::room_message');
+
+class ChatEvent extends ConsumerWidget {
+  final String roomId;
+  final String eventId;
+
+  const ChatEvent({
+    super.key,
+    required this.roomId,
+    required this.eventId,
+  });
+
+  @override
+  Widget build(BuildContext context, WidgetRef ref) {
+    final msg = ref.watch(chatRoomMessageProvider((roomId, eventId)));
+
+    if (msg == null) {
+      _log.severe('Msg not found $roomId $eventId');
+      return const SizedBox.shrink();
+    }
+
+    final inner = msg.eventItem();
+    if (inner == null) {
+      final virtual = msg.virtualItem();
+      if (virtual == null) {
+        _log.severe(
+          'Event is neither virtual nor full event: $roomId $eventId',
+        );
+        return const SizedBox.shrink();
+      }
+      return renderVirtual(msg, virtual);
+    }
+
+    return renderEvent(msg: msg, item: inner, ref: ref);
+  }
+
+  Widget renderVirtual(RoomMessage msg, RoomVirtualItem virtual) {
+    // TODO: virtual Objects support
+    return const SizedBox.shrink();
+  }
+
+  Widget renderEvent({
+    required RoomMessage msg,
+    required RoomEventItem item,
+    required WidgetRef ref,
+  }) {
+    final showAvatar = ref.watch(shouldShowAvatarProvider((roomId, eventId)));
+    final avatarInfo = ref.watch(
+      memberAvatarInfoProvider(
+        (
+          roomId: roomId,
+          userId: item.sender(),
+        ),
+      ),
+    );
+    final options = AvatarOptions.DM(avatarInfo, size: 14);
+    final myId = ref.watch(myUserIdStrProvider);
+    final messageId = msg.uniqueId();
+    final isUser = myId == item.sender();
+    // TODO: render a regular timeline event
+    return Row(
+      mainAxisAlignment:
+          !isUser ? MainAxisAlignment.start : MainAxisAlignment.end,
+      crossAxisAlignment: CrossAxisAlignment.end,
+      mainAxisSize: MainAxisSize.min,
+      children: [
+        (showAvatar && !isUser)
+            ? Padding(
+                padding: const EdgeInsets.only(left: 8),
+                child: ActerAvatar(options: options),
+              )
+            : const SizedBox(width: 40),
+        Flexible(
+          child: _buildEventItem(messageId, item, isUser, showAvatar),
+        ),
+      ],
+    );
+  }
+
+  Widget _buildEventItem(
+    String messageId,
+    RoomEventItem item,
+    bool isUser,
+    bool showAvatar,
+  ) {
+    final eventType = item.eventType();
+    switch (eventType) {
+      case 'm.room.message':
+        return _buildMsgEventItem(roomId, messageId, item, isUser, showAvatar);
+      case 'm.room.redaction':
+        return ChatBubble(
```

Comment:
This is the new UI wrapper implemented for message events inner content. Rest of the implementation is same and copied from old chat widgets.